tests/functional/replication/pauseResumeState.js

```diff
@@ -0,0 +1,550 @@
+            newScheduledDate.setHours(newScheduledDate.getHours() +
+                item.timeChange);
+            // send fake api request
+            mockAPI.resumeCRRService(secondSite, newScheduledDate);
+
+            return async.doWhilst(cb => setTimeout(() => {
+                zkHelper.get(secondSite, (err, data) => {
+                    if (err) {
+                        return cb(err);
+                    }
+                    zkScheduleState = data.scheduledResume;
+                    return cb();
+                });
+            }, 1000), () => {
+                if (zkScheduleState) {
+                    const zkStateDate = new Date(zkScheduleState);
+                    return zkStateDate.getTime() !== newScheduledDate.getTime();
+                }
+                return true;
+            }, err => {
+                assert.ifError(err);
+                assert.strictEqual(new Date(zkScheduleState).toString(),
+                    newScheduledDate.toString());
+                done();
+            });
+        });
+    });
+
+    it('should cancel a scheduled resume when the location is manually ' +
+    'resumed', done => {
+        let zkScheduleState;
+        let zkPauseState;
+        // send fake api request
+        mockAPI.resumeCRRService(secondSite);
+
+        return async.doWhilst(cb => setTimeout(() => {
+            zkHelper.get(secondSite, (err, data) => {
+                if (err) {
+                    return cb(err);
+                }
+                zkPauseState = data.paused;
+                zkScheduleState = data.scheduledResume;
+                return cb();
+            });
+        }, 1000), () => isConsumerActive(consumer2) !== true,
+        err => {
+            assert.ifError(err);
+            assert.strictEqual(zkPauseState, false);
+            assert.strictEqual(zkScheduleState, null);
+            done();
+        });
+    });
+
+    it('should not schedule a resume when the location is already active',
+    done => {
+        // send fake api request
+        mockAPI.resumeCRRService(firstSite, futureDate);
+
+        setTimeout(() => {
+            zkHelper.get(firstSite, (err, data) => {
+                assert.ifError(err);
+
+                assert.strictEqual(data.scheduledResume, null);
+                assert.strictEqual(data.paused, false);
+                assert.strictEqual(isConsumerActive(consumer1), true);
+                done();
+            });
+        }, 1000);
+    });
+
+    it('should resume a location when a scheduled resume triggers',
+    done => {
+        let zkPauseState;
+        let zkScheduleState;
+        // move time forward to trigger scheduled resume
+        const timeMachine = new TimeMachine();
+        timeMachine.install();
+        // hack - for some reason, the scheduled job in QueueProcessor does not
+        // trigger unless I schedule a job here.
+        schedule.scheduleJob(new Date(), () => {});
```

Comment:
I'm not sure why, but when using lolex to move Date forward, the scheduled job in QueueProcessor does not trigger unless I add a separate scheduled job here

Reply:
Possibly could be due to installing after the actual scheduling of the job in QP?